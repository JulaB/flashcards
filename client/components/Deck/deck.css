.deck {
  border-radius: 5px;
  border: 1px solid var(--color-border);

  &__title {
    margin-top: 0;
    text-align: center;
    font-weight: normal;
    background: var(--color-card-bg);
    padding: var(--space-sm);
    color: var(--color-text-light);
  }

  &__info {
    padding: var(--space-sm);
  }
}
