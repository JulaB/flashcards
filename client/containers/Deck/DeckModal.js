import { connect } from 'react-redux';
import CardsSlider from 'components/Card/CardsSlider';

const mapStateToProps = ({ deckReducer: state }, { deckId }) => ({
  cards: state.cards[deckId] || [],
});

export default connect(mapStateToProps)(CardsSlider);
