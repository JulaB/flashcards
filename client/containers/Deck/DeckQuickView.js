import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Loader from 'components/Loader/Loader';
import Button from 'components/Shared/Button/Button';
import { deckFetchData as fetchData } from 'actions/deckActions';
import { showModal } from 'actions/modalActions';
import { DECK_MODAL } from 'constants/modalConstants';

export class DeckQuickViewContainer extends React.Component {
  constructor(props) {
    super(props);
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick() {
    const { isFetching, deckId } = this.props;
    if (isFetching) {
      return;
    }
    this.props.openModalFor(deckId);
  }

  render() {
    const { isFetching } = this.props;
    return (
      <Button
        className="btn--primary deck__button"
        onClick={this.handleClick}
        disabled={isFetching}
      >
        { isFetching
          ? <Loader className="loader--btn" />
          : 'Quick View'
        }
      </Button>
    );
  }
}

DeckQuickViewContainer.propTypes = {
  deckId: PropTypes.string.isRequired,
  name: PropTypes.string,
  isFetching: PropTypes.bool,
  openModalFor: PropTypes.func,
};

DeckQuickViewContainer.defaultProps = {
  isFetching: false,
  name: '',
  openModalFor: () => {},
};

const mapStateToProps = ({ deckReducer: state }, ownProps) => ({
  isFetching: state.deckIsFetching[ownProps.deckId] || false,
});

const mapDispatchToProps = (dispatch, ownProps) => ({
  openModalFor: deckId => (
    dispatch(fetchData(deckId))
      .then(() => dispatch(
        showModal({
          modalType: DECK_MODAL,
          modalProps: {
            deckId,
            open: true,
            title: ownProps.name,
          },
        }),
      ))
  ),
});

export default connect(mapStateToProps, mapDispatchToProps)(DeckQuickViewContainer);
