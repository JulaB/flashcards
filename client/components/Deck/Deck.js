import React from 'react';
import PropTypes from 'prop-types';
import Pluralize from 'components/Shared/Pluralize';
import DeckCardsContainer from 'containers/DeckCards/DeckCards';
import './deck.css';

const Deck = ({ name, id, ...props }) => (
  <div className="deck">
    <h4 className="deck__title">{name}</h4>
    <div className="deck__info">
      <DeckCardsContainer deckId={id} />
      <Pluralize singular='card' plural='cards' count={props['cards-count']} />
    </div>
  </div>
);

Deck.propTypes = {
  id: PropTypes.string,
  name: PropTypes.string,
  'cards-count': PropTypes.number,
};

Deck.defaultProps = {
  id: '0',
  name: '',
  'cards-count': 0,
};
export default Deck;
