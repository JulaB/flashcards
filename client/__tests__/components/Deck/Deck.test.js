import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Deck from 'components/Deck/Deck';

describe('<Deck />', () => {
  it('renders the component', () => {
    const wrapper = shallow(<Deck name="test deck" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
