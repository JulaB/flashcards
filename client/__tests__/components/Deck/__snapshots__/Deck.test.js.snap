// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Deck /> renders the component 1`] = `
<div
  className="deck"
>
  <h4
    className="deck__title"
  >
    test deck
  </h4>
  <div
    className="deck__info"
  >
    <Connect(DeckCards)
      deckId="0"
    />
    <Pluralize
      className=""
      count={0}
      plural="cards"
      singular="card"
    />
  </div>
</div>
`;
